### DISCUSSION TITLE
make mania SVs boundless

### DISCUSSION THREAD

**Discussion by LastExceed** [ppy/osu#13158] [Ideas]
currently SVs have to be in range x0.1 - x10 , which is a good sanity for slowjams/fastjams (which i believe was the original purpose of SVs). However these days SVs are mostly used for teleports and shutters, where both the upper and the lower bound can be quite limiting. Players sometimes work around this by using BPM changes instead, but that is a hacky mess which is hard to maintain in the editor, often requires external editing of the beatmap file, and of course unrankable. 

If possible, negative values should be supported for reverse scrolling effects, but at minimum x0 should be possible for freeze-like effects and the upper bound should be x1000 at least for artifact-free teleports.

One could argue that we're misusing SVs here, but the (granted, better) alternative would be to move these effects to story boarding which would be way more work with barely any advantage compared to just changing the bounds.

**Comment by smoogipoo** [comment]
I don't always agree with the effects people make out of -ve/0 SVs but a lot of people have wanted it over the years, and I think it would be good to add.

Has to be added for all rulesets if it happens, though.

**Comment by eclecticfluff** [comment]
I haven't been able to play OSU yet, but love gimmicks, so hopefully my two cents are relevant enough. While uncapped upper bounds for SV would allow for teleports to be artifact free, I think a separate feature for that might be better. Having teleports be treated separately, and with a field for distance might be more resource efficient, as well as allow for much easier creation of stutters, like that seen in the hi-fi to lo-fi BMS chart for Wavetapper. Still, absurdly high SV might be useful like to have a repeating pattern while the scroll speed increases past unreadability.

On the other subject, I would love negative SVs, though there are several things that would need to be accounted for. Long back-scroll sections like in You Are the Miserable would likely require changes to object loading, so all the notes are visible as they fly backwards. Not to mention, even more extreme cases like World's End BBKKBKK going back and forth over one segment of the chart in space would just add further complications. Furthermore, you know someone (read: me) would want to add notes to be hit during backscroll, like ONGEKI's chart for Red and Blue and Green, which would be practically unreadable mania skins that don't have receptors or in standard. It wouldn't be easy, but if all of that can be dealt with, I would be ecstatic to mess with it.

I suppose at some point gimmick support goes outside the scope of the game, but I hope scroll speed shenanigans are doable, wanted, and ubiquitous enough across rhythm games to warrant the development time.
